test/e2e/testsuites/pre_provisioned_dangling_attach_volume_tester.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/azuredisk"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/azure"
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/credentials"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// PreProvisionedDanglingAttachVolumeTest will provision required PV(s), PVC(s) and Pod(s)
+// Testing that a volume could be reattached to a different node on DanglingAttachError
+type PreProvisionedDanglingAttachVolumeTest struct {
+	CSIDriver       driver.PreProvisionedVolumeTestDriver
+	AzureDiskDriver *azuredisk.Driver
+	Pod             PodDetails
+	VolumeID        string
+	VolumeContext   map[string]string
+}
+
+func (t *PreProvisionedDanglingAttachVolumeTest) Run(client clientset.Interface, namespace *v1.Namespace) {
+
+	// Setup required PV(s) and PVC(s) corresponding to PodDetails that will be shared among two pods
+	tpod, cleanup := t.Pod.SetupWithPreProvisionedVolumes(client, namespace, t.CSIDriver, t.VolumeContext)
+	// Defer must be called here for PV(s) and PVC(s) to be removed after the test completion and not earlier
+	for i := range cleanup {
+		defer cleanup[i]()
+	}
+
+	// Get diskURI, diskName and disk's resourceGroup
+	pvcname := tpod.pod.Spec.Volumes[0].VolumeSource.PersistentVolumeClaim.ClaimName
+	diskURI, err := tpod.GetVolumeDiskURI(pvcname)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting uri for azuredisk %v", err))
+	diskName, err := azuredisk.GetDiskName(diskURI)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting diskName for azuredisk %v", err))
+	resourceGroup, err := azuredisk.GetResourceGroupFromURI(diskURI)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting resourceGroup for azuredisk %v", err))
+
+	// Get azureClient
+	creds, err := credentials.CreateAzureCredentialFile()
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	// Get VMSS
+	vmss, err := azureClient.ListVMSS(context.Background(), resourceGroup)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting the vmss list: %v", err))
```

Comment:
Also check that `len(nodes) > 0` here to avoid possible out-of-bounds error on line 75 if empty list is returned.

You could probably just make the check `>= 2` since we need at least two nodes for this test.